### PR TITLE
:hammer: Add `typing.py` and `env_detection.py` to `mypy` Type Check

### PR DESCRIPTION
**Pull request by Jiaqi-Lv**
- Fix type error in `tiatoolbox/typing.py` and `tiatoolbox/utils/env_detection.py`

### CODE REVIEW

File: tiatoolbox/utils/env_detection.py [hunk 1]
```diff
@@ -365,7 +362,7 @@ def pixman_versions() -> tuple[list, str]:
             determined.
 
     """
-    versions = []
+    versions: list[Any] = []
```
Comment (shaneahmed) [inline tiatoolbox/utils/env_detection.py:365]: shouldn't this be list[int] ? as it says in docstring?

Comment (Jiaqi-Lv) [reply tiatoolbox/utils/env_detection.py:365]: Yeah it should be list[int]

File: tiatoolbox/typing.py [hunk 2]
```diff
@@ -1,24 +1,26 @@
 """Define Variable types for TIAToolbox."""
-from typing import Callable, Literal, Sequence, SupportsFloat
+from typing import Callable, Dict, List, Literal, Sequence, SupportsFloat, Tuple, Union
```
Comment (shaneahmed) [inline tiatoolbox/typing.py:2]:
```suggestion
from __future__ import annotations
from typing import Callable, Dict, List, Literal, Sequence, SupportsFloat, Tuple, Union
```

File: tiatoolbox/utils/env_detection.py [hunk 3]
```diff
@@ -344,12 +340,12 @@ def check_pixman_using_macports(versions: list) -> tuple[list, str]:
         flags=re.MULTILINE,
     )
     if matches:
-        versions = [version_to_tuple(matches.group(1))]
+        versions = [version_to_tuple(match) for match in matches]
```
Comment (Jiaqi-Lv) [inline tiatoolbox/utils/env_detection.py:343]: `re.finall()` returns a list of strings, not a `match` object, so it doesn't have the  `group` attribute.

**Comment by Jiaqi-Lv** [comment]
It seems like we **have** to use `Union` instead of `|` in type annotations, otherwise Python 3.8 and Python 3.9 won't pass the check, (even with `from __future__ import annotations`).

**Review by shaneahmed — APPROVED** [review]
Thanks @Jiaqi-Lv

**Pull request merged**